duckdb-vortex/src/vortex_extension.cpp

```diff
@@ -106,12 +112,22 @@ static void ExtractVortexSchema(const DType *file_dtype, vector<LogicalType> &co
 		std::string field_name(name_buffer, name_len);
 
 		DType *field_dtype = DType_field_dtype(file_dtype, idx);
-		LogicalType duckdb_type = VortexTypeToDuckDBType(DType_get(field_dtype));
+		auto duckdb_type = reinterpret_cast<LogicalType *>(DType_to_duckdb_logical_type(field_dtype));
 
 		column_names.push_back(field_name);
-		column_types.push_back(duckdb_type);
+		column_types.push_back(*duckdb_type);
+		 DType_free(field_dtype);
+	}
+}
 
-		DType_free(field_dtype);
+std::string EnsureFileProtocol(const std::string &path) {
+	const std::string prefix = "file://";
+
+	// Check if the string already starts with "file://"
+	if (path.size() >= prefix.size() && std::equal(prefix.begin(), prefix.end(), path.begin())) {
```

Comment:
Maybe we can switch to a higher C++ version in our extension later and go with: `path.starts_with(prefix)`

Reply:
can we not now?

Reply:
i'll open a separate pr :)